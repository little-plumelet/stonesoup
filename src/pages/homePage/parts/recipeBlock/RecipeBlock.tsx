import { Spin, Typography } from 'antd';
import { RecipeCardList } from '../../../../components/recipeCardList';
import { Recipe } from '../../../../interfaces/recipe.interface';
import style from './style.module.css';

const { Title } = Typography;

interface IRecipeBlockProps {
  title: string;
  recipes: Recipe[];
  loading: boolean;
  error: string | null | undefined;
}
export function RecipeBlock({
  title,
  recipes,
  loading,
  error,
}: IRecipeBlockProps) {
  return (
    <div>
      <Title level={3}>{title}</Title>
      {loading && (
        <div className={style.spinContainer}>
          <Spin size="large" data-testid="spin" />
        </div>
      )}
      {error && (
        <p data-testid="error" className={style.error}>
          {error}
        </p>
      )}
      {!!recipes.length && <RecipeCardList recipes={recipes} />}
    </div>
  );
}
