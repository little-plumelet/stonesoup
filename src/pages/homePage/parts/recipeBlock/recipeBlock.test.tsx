import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { RecipeBlock } from './RecipeBlock';
import '../../../../../matchMedia';

const mockRecipeList = [
  {
    id: 716428,
    title: 'Pasta with Garlic, Scallions, Cauliflower & Breadcrumbs',
    image: 'https://spoonacular.com/recipeImages/716429-556x370.jpg',
    readyInMinutes: 15,
    aggregateLikes: 29,
  },
  {
    id: 716429,
    title: 'Risotto with mashrums',
    image: 'https://spoonacular.com/recipeImages/716429-556x370.jpg',
    readyInMinutes: 57,
    aggregateLikes: 84,
  },
];

describe('recipeBlock', () => {
  it('should render', async () => {
    const component = render(
      <MemoryRouter>
        <RecipeBlock
          title="title"
          error={null}
          loading={false}
          recipes={mockRecipeList}
        />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
    expect(
      await screen.findByText('Risotto with mashrums')
    ).toBeInTheDocument();
  });

  it('should render loading', async () => {
    render(
      <MemoryRouter>
        <RecipeBlock title="title" error={null} loading recipes={[]} />
      </MemoryRouter>
    );

    expect(await screen.findByTestId('spin')).toBeInTheDocument();
  });

  it('should render error', async () => {
    render(
      <MemoryRouter>
        <RecipeBlock title="title" error="404" loading={false} recipes={[]} />
      </MemoryRouter>
    );

    expect(await screen.findByTestId('error')).toBeInTheDocument();
  });
});
