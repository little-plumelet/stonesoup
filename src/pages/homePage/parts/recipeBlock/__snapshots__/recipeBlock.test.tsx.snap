// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipeBlock should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="ant-typography css-dev-only-do-not-override-mxhywb"
      >
        title
      </h3>
      <div
        class="ant-row ant-row-start css-dev-only-do-not-override-mxhywb"
        style="margin: -16px -12px -16px -12px;"
      >
        <div
          class="ant-col ant-col-6 ant-col-xs-24 ant-col-sm-12 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4 css-dev-only-do-not-override-mxhywb"
          style="padding: 16px 12px 16px 12px;"
        >
          <div
            class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-mxhywb"
          >
            <div
              class="ant-card-cover"
            >
              <img
                alt="recipe"
                src="https://spoonacular.com/recipeImages/716429-556x370.jpg"
              />
            </div>
            <div
              class="ant-card-body"
            >
              <div
                class="ant-card-meta"
              >
                <div
                  class="ant-card-meta-detail"
                >
                  <div
                    class="ant-card-meta-title"
                  >
                    <span>
                      Pasta with Garlic, Scallions, Cauliflower & Breadcrumbs
                    </span>
                  </div>
                  <div
                    class="ant-card-meta-description"
                  >
                    <div
                      class="cookingTimeBlock"
                    >
                      <span
                        aria-label="dashboard"
                        class="anticon anticon-dashboard cookingTimeIcon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="dashboard"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                      <span
                        class="cookingTime"
                      >
                        cooking time - 15 min
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <footer
                class="footer"
              >
                <div
                  class="likeBlock"
                >
                  <span
                    class="like"
                  >
                    29
                  </span>
                  <span
                    aria-label="like"
                    class="anticon anticon-like"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="like"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M273 495.9v428l.3-428zm538.2-88.3H496.8l9.6-198.4c.6-11.9-4.7-23.1-14.6-30.5-6.1-4.5-13.6-6.8-21.1-6.7-19.6.1-36.9 13.4-42.2 32.3-37.1 134.4-64.9 235.2-83.5 302.5V852h399.4a56.85 56.85 0 0033.6-51.8c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-14-25.5 21.9-19a56.76 56.76 0 0019.6-43c0-19.1-11-37.5-28.8-48.4z"
                        fill="#e6f7ff"
                      />
                      <path
                        d="M112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32zm773.9 5.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.5-65.5-111a67.67 67.67 0 00-34.3-9.3H572.3l6-122.9c1.5-29.7-9-57.9-29.5-79.4a106.4 106.4 0 00-77.9-33.4c-52 0-98 35-111.8 85.1l-85.8 310.8-.3 428h472.1c9.3 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37zM820.4 499l-21.9 19 14 25.5a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H345V506.8c18.6-67.2 46.4-168 83.5-302.5a44.28 44.28 0 0142.2-32.3c7.5-.1 15 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.1 32.2-19.6 43z"
                        fill="#1890ff"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="buttonBlock"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-default"
                    type="button"
                  >
                    <span>
                      View recipe
                    </span>
                  </button>
                </div>
              </footer>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-6 ant-col-xs-24 ant-col-sm-12 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4 css-dev-only-do-not-override-mxhywb"
          style="padding: 16px 12px 16px 12px;"
        >
          <div
            class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-mxhywb"
          >
            <div
              class="ant-card-cover"
            >
              <img
                alt="recipe"
                src="https://spoonacular.com/recipeImages/716429-556x370.jpg"
              />
            </div>
            <div
              class="ant-card-body"
            >
              <div
                class="ant-card-meta"
              >
                <div
                  class="ant-card-meta-detail"
                >
                  <div
                    class="ant-card-meta-title"
                  >
                    <span>
                      Risotto with mashrums
                    </span>
                  </div>
                  <div
                    class="ant-card-meta-description"
                  >
                    <div
                      class="cookingTimeBlock"
                    >
                      <span
                        aria-label="dashboard"
                        class="anticon anticon-dashboard cookingTimeIcon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="dashboard"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                      <span
                        class="cookingTime"
                      >
                        cooking time - 57 min
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <footer
                class="footer"
              >
                <div
                  class="likeBlock"
                >
                  <span
                    class="like"
                  >
                    84
                  </span>
                  <span
                    aria-label="like"
                    class="anticon anticon-like"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="like"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M273 495.9v428l.3-428zm538.2-88.3H496.8l9.6-198.4c.6-11.9-4.7-23.1-14.6-30.5-6.1-4.5-13.6-6.8-21.1-6.7-19.6.1-36.9 13.4-42.2 32.3-37.1 134.4-64.9 235.2-83.5 302.5V852h399.4a56.85 56.85 0 0033.6-51.8c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-14-25.5 21.9-19a56.76 56.76 0 0019.6-43c0-19.1-11-37.5-28.8-48.4z"
                        fill="#e6f7ff"
                      />
                      <path
                        d="M112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32zm773.9 5.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.5-65.5-111a67.67 67.67 0 00-34.3-9.3H572.3l6-122.9c1.5-29.7-9-57.9-29.5-79.4a106.4 106.4 0 00-77.9-33.4c-52 0-98 35-111.8 85.1l-85.8 310.8-.3 428h472.1c9.3 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37zM820.4 499l-21.9 19 14 25.5a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H345V506.8c18.6-67.2 46.4-168 83.5-302.5a44.28 44.28 0 0142.2-32.3c7.5-.1 15 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.1 32.2-19.6 43z"
                        fill="#1890ff"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="buttonBlock"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-default"
                    type="button"
                  >
                    <span>
                      View recipe
                    </span>
                  </button>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h3
      class="ant-typography css-dev-only-do-not-override-mxhywb"
    >
      title
    </h3>
    <div
      class="ant-row ant-row-start css-dev-only-do-not-override-mxhywb"
      style="margin: -16px -12px -16px -12px;"
    >
      <div
        class="ant-col ant-col-6 ant-col-xs-24 ant-col-sm-12 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4 css-dev-only-do-not-override-mxhywb"
        style="padding: 16px 12px 16px 12px;"
      >
        <div
          class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-mxhywb"
        >
          <div
            class="ant-card-cover"
          >
            <img
              alt="recipe"
              src="https://spoonacular.com/recipeImages/716429-556x370.jpg"
            />
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-card-meta"
            >
              <div
                class="ant-card-meta-detail"
              >
                <div
                  class="ant-card-meta-title"
                >
                  <span>
                    Pasta with Garlic, Scallions, Cauliflower & Breadcrumbs
                  </span>
                </div>
                <div
                  class="ant-card-meta-description"
                >
                  <div
                    class="cookingTimeBlock"
                  >
                    <span
                      aria-label="dashboard"
                      class="anticon anticon-dashboard cookingTimeIcon"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="dashboard"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                    <span
                      class="cookingTime"
                    >
                      cooking time - 15 min
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <footer
              class="footer"
            >
              <div
                class="likeBlock"
              >
                <span
                  class="like"
                >
                  29
                </span>
                <span
                  aria-label="like"
                  class="anticon anticon-like"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="like"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M273 495.9v428l.3-428zm538.2-88.3H496.8l9.6-198.4c.6-11.9-4.7-23.1-14.6-30.5-6.1-4.5-13.6-6.8-21.1-6.7-19.6.1-36.9 13.4-42.2 32.3-37.1 134.4-64.9 235.2-83.5 302.5V852h399.4a56.85 56.85 0 0033.6-51.8c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-14-25.5 21.9-19a56.76 56.76 0 0019.6-43c0-19.1-11-37.5-28.8-48.4z"
                      fill="#e6f7ff"
                    />
                    <path
                      d="M112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32zm773.9 5.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.5-65.5-111a67.67 67.67 0 00-34.3-9.3H572.3l6-122.9c1.5-29.7-9-57.9-29.5-79.4a106.4 106.4 0 00-77.9-33.4c-52 0-98 35-111.8 85.1l-85.8 310.8-.3 428h472.1c9.3 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37zM820.4 499l-21.9 19 14 25.5a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H345V506.8c18.6-67.2 46.4-168 83.5-302.5a44.28 44.28 0 0142.2-32.3c7.5-.1 15 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.1 32.2-19.6 43z"
                      fill="#1890ff"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="buttonBlock"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-default"
                  type="button"
                >
                  <span>
                    View recipe
                  </span>
                </button>
              </div>
            </footer>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-6 ant-col-xs-24 ant-col-sm-12 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4 css-dev-only-do-not-override-mxhywb"
        style="padding: 16px 12px 16px 12px;"
      >
        <div
          class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-mxhywb"
        >
          <div
            class="ant-card-cover"
          >
            <img
              alt="recipe"
              src="https://spoonacular.com/recipeImages/716429-556x370.jpg"
            />
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-card-meta"
            >
              <div
                class="ant-card-meta-detail"
              >
                <div
                  class="ant-card-meta-title"
                >
                  <span>
                    Risotto with mashrums
                  </span>
                </div>
                <div
                  class="ant-card-meta-description"
                >
                  <div
                    class="cookingTimeBlock"
                  >
                    <span
                      aria-label="dashboard"
                      class="anticon anticon-dashboard cookingTimeIcon"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="dashboard"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                    <span
                      class="cookingTime"
                    >
                      cooking time - 57 min
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <footer
              class="footer"
            >
              <div
                class="likeBlock"
              >
                <span
                  class="like"
                >
                  84
                </span>
                <span
                  aria-label="like"
                  class="anticon anticon-like"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="like"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M273 495.9v428l.3-428zm538.2-88.3H496.8l9.6-198.4c.6-11.9-4.7-23.1-14.6-30.5-6.1-4.5-13.6-6.8-21.1-6.7-19.6.1-36.9 13.4-42.2 32.3-37.1 134.4-64.9 235.2-83.5 302.5V852h399.4a56.85 56.85 0 0033.6-51.8c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-13.9-25.4 21.9-19a56.76 56.76 0 0019.6-43c0-9.7-2.3-18.9-6.9-27.3l-14-25.5 21.9-19a56.76 56.76 0 0019.6-43c0-19.1-11-37.5-28.8-48.4z"
                      fill="#e6f7ff"
                    />
                    <path
                      d="M112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32zm773.9 5.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.5-65.5-111a67.67 67.67 0 00-34.3-9.3H572.3l6-122.9c1.5-29.7-9-57.9-29.5-79.4a106.4 106.4 0 00-77.9-33.4c-52 0-98 35-111.8 85.1l-85.8 310.8-.3 428h472.1c9.3 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37zM820.4 499l-21.9 19 14 25.5a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.1 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H345V506.8c18.6-67.2 46.4-168 83.5-302.5a44.28 44.28 0 0142.2-32.3c7.5-.1 15 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.1 32.2-19.6 43z"
                      fill="#1890ff"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="buttonBlock"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-default"
                  type="button"
                >
                  <span>
                    View recipe
                  </span>
                </button>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
